### ISSUE TITLE
IE / Edge: Unable to get property 'x' of undefined or null reference

### ISSUE THREAD

**Issue by hyperknot** [mapbox/mapbox-gl-js#6800]
**mapbox-gl-js version**: 0.45.0
**browser**: all kind of Microsoft browsers: IE 11, Edge 16-17, Edge Mobile 14-15 

No idea what triggers it, I believe it's as simple as clicking on the canvas when a 

```
map.on('click', ...)
``` 

is registered.

Traceback:
https://sentry.io/share/issue/b5ae207185f2486ea311e4a63a37c39f/

**Comment by jfirebaugh** [comment]
Thanks for the report @hyperknot, but I'm afraid unless you have more information about how to trigger it, this is not going to be actionable for us.

**Comment by hyperknot** [comment]
OK, there is one single report from Firefox 60, about the same error. In Firefox, the error message is:

```
e is null
```

I think the issue is actually quite simple. 

1. In some cases `bind_handlers.js/onMouseDown` does not initialise `startPos`, for example because the event is actually a touch event.
2. Then onClick is fired, trying to compare with `null.x`, which doesn't work, rightfully so.

I guess the logic needs to be revised about how to handle touch based events + onClick. + also a simple check like 

```
if (startPost && pos.equals(startPos)) {
    map.fire(new MapMouseEvent('click', map, e));
}
```

would be needed.

**Comment by hyperknot** [comment]
Anyone who can try to test this on windows touch devices?

**Comment by mourner** [comment]
@hyperknot hey, would you like to submit a PR for this?

**Comment by hyperknot** [comment]
I cannot, as I have Windows devices with a touch screen. If someone has
Window device with a touch screen feel free to look at that.

**Comment by mourner** [comment]
This was likely fixed by #8462 and is a duplicate of #7864.
